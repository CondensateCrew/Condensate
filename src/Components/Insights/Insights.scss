@import '../../css/variables.scss';
@import '../../css/index.scss';

