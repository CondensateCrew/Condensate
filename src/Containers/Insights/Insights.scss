@import '../../css/variables.scss';
@import '../../css/index.scss';

.insights-div {
  width: 90%;
  height: 5vh;
  display: flex;
  margin: $medium-space;
  padding: $medium-space;
  justify-content: space-between;
  align-items: center;
  border-radius: $smallest-border-radius;
  img {
    width: 1rem;
  }
  h3 {
    font-size: 1rem;
  }
  box-shadow: $external-shadow;
}