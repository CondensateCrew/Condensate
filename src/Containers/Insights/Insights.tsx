import React from 'react';
import Close from '../../assets/close.svg';
import './Insights.scss';

interface Props {
  text: string
}

const Insights:React.FC<Props> = ({ text }) => {
  return (
    <div className='insights-div'>
      <h3>{text}</h3>
      <img src={Close} alt='delete icon' />
    </div>
  )
}

export default Insights;