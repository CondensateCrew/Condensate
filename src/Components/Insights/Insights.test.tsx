/**
 * @jest-environment jsdom
 */

import React from 'react';
import Insights from './Insights';
import { mount } from 'enzyme';

describe('Insights component', () => {
  let wrapper: any;

  const mockSetResponses = jest.fn();

  const mockProps = {
    text: 'Response',
    responses: ['Response'],
    setResponses: mockSetResponses
  };

  beforeEach(() => {
    wrapper = mount(<Insights {...mockProps}/>);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });


  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call setResponses after remove button is clicked', () => {
    wrapper.find('img').simulate('click');
    expect(mockSetResponses).toHaveBeenCalled();
  });
});
