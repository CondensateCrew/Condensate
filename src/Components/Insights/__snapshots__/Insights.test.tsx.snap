// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Insights component should match the snapshot 1`] = `
<Insights
  responses={
    Array [
      "Response",
    ]
  }
  setResponses={[MockFunction]}
  text="Response"
>
  <div
    className="insights-div"
  >
    <h3>
      Response
    </h3>
    <img
      alt="delete icon"
      onClick={[Function]}
      src="close.svg"
    />
  </div>
</Insights>
`;
